specification/dsdl/expression_types.tex

```diff
@@ -0,0 +1,223 @@
+
+\section{Expression types}\label{sec:dsdl_expression_types}
+
+Expression types are a special category of data types whose instances can only exist and be operated upon
+at the time of DSDL definition processing.
+As such, expression types cannot be used to define attributes,
+and their instances cannot be exchanged between nodes.
+
+Expression types are used to represent values of constant expressions which are evaluated
+when a DSDL definition is processed.
+Results of such expressions can be used to define various constant properties,
+such as array length boundaries or values of constant attributes.
+
+Expression types are specified in this section.
+Each expression type has a formal DSDL name for completeness;
+even if such types can't be used to define attributes,
+a well-defined formal name allows DSDL processing tools to emit well-formed
+and understandable diagnostic messages.
+
+\subsection{Rational number}\label{sec:dsdl_rational}
+
+At the time of DSDL definition processing, integer and real numbers are represented internally as rational numbers
+where the range of numerator and denominator is unlimited\footnote{%
+Technically, the range may only be limited by the memory resources available to the DSDL processing tool.
+DSDL processing tools are not permitted to introduce any implicit rational number transformations that
```

Comment:
This is a requirement of the specification  and should be pulled out of a footnote and placed into a normative section in the text.